src/main/java/net/neoforged/neoforge/common/NeoForgeMod.java

```diff
@@ -601,6 +605,8 @@ public NeoForgeMod(IEventBus modEventBus, Dist dist, ModContainer container) {
         UsernameCache.load();
         if (dist.isClient()) ClientCommandHandler.init();
         DualStackUtils.initialise();
+        TagConventionLogWarning.init();
+        if (FMLEnvironment.dist == Dist.CLIENT) TagConventionLogWarningClient.init();
```

Comment:
The dist is a ctor parameter, and you can probably merge this with the `if` above.